docs/architecture/adr-074-implicit-msg-signers.md

```diff
@@ -0,0 +1,126 @@
+defining who the signer of a message was and its implementation in the `x/tx` module which
+extracts signers dynamically and allowed removing the dependency on the global bech32
+configuration.
+
+Still this design introduces a fair amount of complexity. For instance, inter-module message
+passing ([ADR 033](./adr-033-protobuf-inter-module-comm.md)) has been in discussion for years
+without much progress and one of the main blockers is figuring out how to properly authenticate
+messages in a performant and consistent way. With embedded message signers there will always need
+to be a step of extracting the signer and then checking with the module sending is actually
+authorized to perform the operation. With dynamic signer extraction, although the system is
+more consistent, more performance overhead is introduced. In any case why should an inter-module
+message passing system need to do so much conversion, parsing, etc. just to check if a message
+is authenticated? In addition, we have the complexity where modules can actually have many valid
+addresses. How are we to accommodate this? Should there be a lookup into `x/auth` to check if an
+address belongs to a module or not? All of these thorny questions are delaying the delivery of
+inter-module message passing because we do not want an implementation that is overly complex.
+There are many use cases for inter-module message passing which are still relevant, the most
+immediate of which is a more robust denom management system in `x/bank` `v2` which is being explored
+in [ADR 071](https://github.com/cosmos/cosmos-sdk/pull/20316).
+
+## Alternatives
+
+Alternatives that have been considered are extending the current `x/tx` signer extraction system
+to inter-module message passing as defined in [ADR 033](./adr-033-protobuf-inter-module-comm.md).
+
+## Decision
+
+We have decided to introduce a new `MsgV2` standard whereby the signer of the message is implied
+by the credentials of the party sending it. These messages will be distinct from the existing messages
+and define new semantics with the understanding that signers are implicit.
+
+In the case of messages passed internally by a module or `x/account` instance, the signer of a message
+will simply be the main root address of the module or account sending the message. An interface for
+safely passing such messages to the message router will need to be defined.
+
+In the case of messages passed externally in transactions, `MsgV2` instances will need to be wrapped
+in a `MsgV2` envelope:
+```protobuf
+message MsgV2 {
+  string signer = 1;
+  google.protobuf.Any msg = 2;  
+}
+```
+
+Because the `cosmos.msg.v1.signer` annotation is required currently, `MsgV2` types should set the message option
+`cosmos.msg.v2.is_msg` to `true` instead.
+
+Here is an example comparing a v1 an v2 message:
+```protobuf
+// v1
+message MsgSendV1 {
+  string from_address = 1 [(cosmos.msg.v1.signer) = true];
```

Comment:
Does this work? `cosmos.msg.v1.signer` is a message annotation of type string, not a field annotation of type bool right